src/tool_formparse.c

```diff
@@ -372,16 +383,146 @@ static int get_param_part(struct OperationConfig *config, char **str,
   return sep & 0xFF;
 }
 
-/* Check if file is "-". If so, use a callback to read OUR stdin (to
+
+/* Mime part callbacks for stdin. */
+static size_t stdin_read(char *buffer, size_t size, size_t nitems, void *arg)
```

Comment:
There's already a [file2memory](https://github.com/curl/curl/blob/master/src/tool_paramhlp.c#L90) function you should use instead!